internal/gengapic/gengapic.go

```diff
@@ -348,6 +348,30 @@ func (g *generator) insertRequestHeaders(m *descriptor.MethodDescriptorProto, t
 	}
 }
 
+// insertAutoPopulatedFields generates the conditional initialization of any
+// default-value request fields (for the given method) that are specified for
+// auto-population by autoPopulatedFields.
+//
+// If the field value is not equal to default value at the time of sending the
+// request, implying it was set by the user, or if the field has explicit
+// presence and is set by the user, the field must not be auto-populated by
```

Comment:
Is there currently any way with Go protos to determine explicit presence?

Reply:
The `FieldDescriptorProto` should have an api for that `GetProto3Optional`

https://pkg.go.dev/google.golang.org/protobuf@v1.32.0/types/descriptorpb#FieldDescriptorProto.GetProto3Optional

Reply:
If `GetProto3Optional()` is `true`, then the field will be a pointer e.g. `*string` in the Go code, so you can generate some code like `if req.RequestId == nil { /*autopopulate*/ }`

yes I know that code uses a direct field access instead of a getter, but this is the only way to check presence with protobuf-go open structs :)

Reply:
Thank you, will do!

Reply:
Looks like I added a helper for this: https://github.com/googleapis/gapic-generator-go/blob/e446700543fd12d2c3c479263c532f7e4bbeee0a/internal/gengapic/helpers.go#L84-L94

Reply:
Updated to use `req.RequestId == nil`, can you please verify that this looks correct in this func and in the `want` files below?

Reply:
OK, I won't rely on the field simply not being REQUIRED, I'll add the `isOptional` check as well, thanks.

Reply:
> OK, I won't rely on the field simply not being REQUIRED, I'll add the `isOptional` check as well, thanks.

`google.api.field_behavior` doesn't actually indicate anything about field presence - `proto3` using the label `optional` for field presence was kind of awkward. So IDK if the `REQUIRED` check is even necessary unless the design states as much.

Reply:
Oh actually the label `required` (which you are checking) is only for `proto2`. Definitely don't need that

Reply:
Fixed, thanks!